lib/decompress/huf_decompress_impl.h

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (c) 2018-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef FUNCTION_NAME
+#  error "FUNCTION_NAME(name) must be defined"
+#endif
+
+#ifndef TARGET
+#  error "TARGET must be defined"
+#endif
+
+
+static TARGET BYTE FUNCTION_NAME(HUF_decodeSymbolX2)(BIT_DStream_t* Dstream, const HUF_DEltX2* dt, const U32 dtLog)
```

Comment:
One suggestion : 
would that produce the same optimization with a construction like that : 

```
static type HUF_decodeX() {  (...) }   // normal function, default compilation flags

#ifdef DYNAMIC_BMI2
static TARGET_ATTRIBUTE("bmi2") type HUF_decodeX_bmi2() {
   return HUF_decodeX();   // Ensure it's inlined
}
#endif
```

It would avoid creation of an additional file to include several times, which is also harder to parse for editors.
But it's unclear if it would produce the same gains.

Reply:
I can try it

Reply:
Sadly I didn't get the same gains using that method.